src/agent/actions/prompt.ts

```diff
@@ -0,0 +1,145 @@
+import {type Observable} from 'rxjs'
+
+import {_request} from '../../data/dataMethods'
+import type {ObservableSanityClient, SanityClient} from '../../SanityClient'
+import type {AgentActionParams, Any, HttpRequest} from '../../types'
+import {hasDataset} from '../../validators'
+
+/**  @beta */
+export interface PromptRequestBase {
+  /**
+   * Instruct the LLM how it should generate content. Be as specific and detailed as needed.
+   *
+   * The LLM only has access to information in the instruction, plus the target schema.
+   *
+   * string template using $variable
```

Comment:
> string template using $variable

This does read a little oddly to me, this could possibly be a little more explicit, as `$variable` usage isn't required. Will defer to @markmichon here.

Reply:
Changing to String template with support for $variable from `instructionParams`.